src/__tests__/role.js

```diff
@@ -183,6 +184,138 @@ test('can include inaccessible roles', () => {
   expect(getByRole('list', {hidden: true})).not.toBeNull()
 })
 
+test('can be filtered by accessible name', () => {
+  const {getByRole} = renderIntoDocument(
+    `
+<div>
+  <h1>Order</h1>
+  <h2>Delivery Adress</h2>
+  <form aria-label="Delivery Adress">
+    <label>
+      <div>Street</div>
+      <input type="text" />
+    </label>
+    <input type="submit" />
+  </form>
+  <h2>Invoice Adress</h2>
+  <form aria-label="Invoice Adress">
+    <label>
+      <div>Street</div>
+      <input type="text" />
+    </label>
+    <input type="submit" />
+  </form>
+</div>`,
+  )
+
+  const deliveryForm = getByRole('form', {name: 'Delivery Adress'})
+  expect(deliveryForm).not.toBeNull()
+
+  expect(
+    // TODO: upstream bug in `aria-query`; should be `button` role
+    getQueriesForElement(deliveryForm).getByRole('textbox', {name: 'Submit'}),
+  ).not.toBeNull()
+
+  const invoiceForm = getByRole('form', {name: 'Delivery Adress'})
+  expect(invoiceForm).not.toBeNull()
+
+  expect(
+    getQueriesForElement(invoiceForm).getByRole('textbox', {name: 'Street'}),
+  ).not.toBeNull()
+})
+
+test('accessible name filter implements TextMatch', () => {
+  const {getByRole} = render(
+    `<h1>Sign <em>up</em></h1><h2>Details</h2><h2>Your Signature</h2>`,
+  )
+
+  // subset via regex
+  expect(getByRole('heading', {name: /gn u/})).not.toBeNull()
+  // regex
+  expect(getByRole('heading', {name: /^sign/i})).not.toBeNull()
+  // function
+  expect(
+    getByRole('heading', {
+      name: (name, element) => {
+        return element.nodeName === 'H2' && name === 'Your Signature'
+      },
+    }),
+  ).not.toBeNull()
+})
+
+test('includes accesible names in error message', () => {
+  const {getByRole} = render(`<h1>Sign <em>up</em></h1>`)
+
+  expect(() => getByRole('heading', {name: 'Sign Up'}))
+    .toThrowErrorMatchingInlineSnapshot(`
+"Unable to find an accessible element with the role "heading" and name "Sign Up"
+
+Here are the accessible roles:
+
+  heading:
+
+  Name "Sign up":
```

Comment:
The only difference between this and the query is the `u` is not capitalized. Could we make this difference more stark so this makes more sense why it failed to find an element please? Thanks!

Reply:
I had trouble understanding it as well. I would like to keep the test to make sure that `{ name: string }` is always case-sensitive. I'll add a comment to the test explaining the issue.